src/index.js

```diff
@@ -76,18 +81,22 @@ const extractAuth = (opts) => {
   }
 
   const url = new URL(opts.name);
-  if (!url.username) {
+  if (!url.username && !opts.session) {
     return;
   }
 
   opts.credentials = {
     username: url.username,
     password: url.password
   };
+
+  if (opts.session) {
+    setSession(opts, { token: opts.session, expires: Date.now() + oneYearInMs });
```

Comment:
This is only used internally (the plugin checks the expiration date of the "cookie", and requests a new one before the final request if it's expired). 
I can change this behavior to not check expiration if it's not set at all.

Reply:
That would be fine too, but honestly I think setting it to some constant "end of the universe" number makes the logic clearer because you don't have to add the null check.

Reply:
I set it to Number.MAX_SAFE_INTEGER